python/ray/serve/_private/proxy_state.py

```diff
@@ -32,7 +34,13 @@
 
 class ProxyState:
     def __init__(
-        self, actor_handle: ActorHandle, actor_name: str, node_id: str, node_ip: str
+        self,
+        actor_handle: ActorHandle,
+        actor_name: str,
+        node_id: str,
+        node_ip: str,
+        proxy_restart_count: int = 0,
+        ray_api_wrapper: ModuleType = ray,
```

Comment:
Note: this is now injected so we can use a fake api wrapper in test. By default ray will be used. I didn't bother to create a real api wrapper class just to expose the same interface ray provided, but let me know if you think it's necessary for serve to have a shared ray api wrapper layer.